File: ObjectRequestBrokerJava/src/main/java/examples/info/InfoServer.java
package examples.info;

import orbapi.ORB;

import java.util.Random;

public class InfoServer implements IInfo {
    private final static String INFO_SERVER_NAME = "InfoServer";
    private final static int INFO_SERVER_PORT = 2020;

    @Override
    public String get_road_info(int road_ID) {
        return "Informations about road " + road_ID;
    }

    @Override
    public double get_temp(String city) {
        return new Random().nextDouble();
    }

    public static void main(String[] args) {
        InfoServer infoServer = new InfoServer();
        ORB.register(INFO_SERVER_NAME,INFO_SERVER_PORT,"info",infoServer);
    }
}

File: ObjectRequestBrokerCS/ServerExecutable/InfoServer.cs
using System;
using ORB.orbapi;

namespace ServerExecutable
{
    public class InfoServer:IInfo
    {
        private static readonly string INFO_SERVER_NAME = "InfoServer";
        private static readonly int INFO_SERVER_PORT = 2020;


        public string get_road_info(long road_ID)
        {
            return "Informations about road " + road_ID;
        }


        public double get_temp(string city)
        {
            return new Random().NextDouble();
        }

        public static void start()
        {
            var infoServer = new InfoServer();
            ORBMiddleware.Register(INFO_SERVER_NAME, INFO_SERVER_PORT, "info", infoServer);
        }
    }
}